src/ts/generic/geometry.ts

```diff
@@ -80,26 +81,23 @@ export function flat(
     }
 }
 
-export function parseGeometry(
-    geometry: ISimpleGeometry,
-    type: GeometryType
```

Comment:
So at the moment I simplified this to never accept the provided geometry type, and always rely on the geometry type specified in the feature. I need to research whether this is problematic or not.

It raises the question of whether there are cases where people would prefer to set the type in the Header only and not in the individual features (e.g. if your file has only points, set point in the header and leave type blank on the individual geometries). I need to confirm whether that's actually valid in the format, and if so whether there are any compelling reasons for supporting it.

Reply:
Yes the "normal" is that homogenous collections only have defined type and columns in the header. Default values in Flatbuffers do not need any storage space so there is a good reason for it too. 🙂

Reply:
That makes sense. So in that case we need to still probably support this parameter (rather than removing it as I have done), but the logic would be that if the provided GeometryType is "unknown", then we look at the type of the individual geometry.

Reply:
Yep, I want this done so I've made https://github.com/flatgeobuf/flatgeobuf/issues/161 for it.

Reply:
Sounds good. I updated this branch with the change discussed above. I saw you merged the other PR with the formatter update so in the morning I will rebase this one and then it should be good to go I think.

Reply:
Ok I'll try to get time to review this. I think it at least needs some test cases though.

Reply:
Sounds good. I had added a couple but I think they were hard to see due to the big formatter diff in that spec file. But I added a few more now as well, let me know what you think.

Reply:
Also I see the CI is running now, thanks :+1: